specutils/spectra/spectrum_collection.py

```diff
@@ -124,7 +129,24 @@ def from_spectra(cls, spectra):
 
         # Compose multi-dimensional ndarrays for each property
         flux = u.Quantity([spec.flux for spec in spectra])
-        spectral_axis = u.Quantity([spec.spectral_axis for spec in spectra])
+
+        # Check that the spectral parameters are the same for each input
+        # spectral_axis and create the multi-dimensional SpectralCoord
+        sa = [x.spectral_axis for x in spectra]
+        if not all(x.radial_velocity == sa[0].radial_velocity for x in sa) or \
+            not all(x.target == sa[0].target for x in sa) or \
+            not all(x.observer == sa[0].observer for x in sa) or \
+            not all(x.doppler_convention == sa[0].doppler_convention for
+                    x in sa) or \
+            not all(x.doppler_rest == sa[0].doppler_rest for x in sa):
+                raise ValueError("All input spectral_axis SpectralCoord "
+                                 "objects must have the same parameters.")
+        spectral_axis = SpectralCoord(sa,
```

Comment:
Two problems with the part from here down:
1. In principle it should be OK for an RV/observer/target to be multi-dimensional, matching the dimension of the spectral collection. But that requires a lot of work so probably we should just make an issue and do a follow-on PR?
2. The fact that the "first" one is used should be in the docs (could be just the docstring for now)

Reply:
I'll add a note to the docstring for now and make another issue for this.

Reply:
I added a note to the docstrings that the spectral_axis parameters for each spectrum need to be identical. Given this (and the fact that an error is raised if they aren't) I don't think we need to specifically call out that the first one is used.

Reply:
Hmm.  I have some misgivings about this, but they are all things that make this "less" restrictive, so best bet is to probably deal with this in a follow-on issue so as not to block this PR further.  I'll make a relevant issue.